bot/utils/pagination.py

```diff
@@ -0,0 +1,389 @@
+    * max_lines: `int`
+        The maximum amount of lines allowed in a page.
+    """
+
+    def __init__(
+        self,
+        prefix: str = "```",
+        suffix: str = "```",
+        max_size: int = 2000,
+        scale_to_size: int = 2000,
+        max_lines: t.Optional[int] = None,
+    ) -> None:
+        """
+        This function overrides the Paginator.__init__ from inside discord.ext.commands.
+
+        It overrides in order to allow us to configure the maximum number of lines per page.
+        """
+        self.prefix = prefix
+        self.suffix = suffix
+
+        # Embeds that exceed 2048 characters will result in an HTTPException (Discord API limit)
+        if max_size > 2000:
+            raise ValueError(f"max_size must be <= 2,000 characters. ({max_size})")
+
+        self.max_size = max_size - len(suffix)
+
+        if scale_to_size < max_size:
+            raise ValueError(
+                f"scale_to_size must be >= max_size. ({scale_to_size=}, {max_size=})"
+            )
+
+        if scale_to_size > 2000:
+            raise ValueError(
+                f"scale_to_size must be <= 2,000 characters. ({scale_to_size})"
+            )
+
+        self.scale_to_size = scale_to_size - len(suffix)
+        self.max_lines = max_lines
+        self._current_page = [prefix]
+        self._line_count = 0
+        self._count = len(prefix) + 1  # prefix + newline
+        self._pages = []
+
+    def add_line(self, line: str = "", *, empty: bool = False) -> None:
+        """
+        Adds a line to the current page.
+
+        If a line on a page exceeds `max_size` characters, then `max_size` will go up to
+        `scale_to_size` for a single line before creating a new page for the overflow words. If it
+        is still exceeded, the excess characters are stored and placed on the next pages unti
+        there are none remaining (by word boundary). The line is truncated if `scale_to_size` is
+        still exceeded after attempting to continue onto the next page.
+
+        In the case that the page already contains one or more lines and the new lines would cause
+        `max_size` to be exceeded, a new page is created. This is done in order to make a best
+        effort to avoid breaking up single lines across pages, while keeping the total length of the
+        page at a reasonable size.
+        """
+        remaining_words = None
+        if len(line) > (max_chars := self.max_size - len(self.prefix) - 2):
+            if len(line) > self.scale_to_size:
+                line, remaining_words = self._split_remaining_words(line, max_chars)
+                if len(line) > self.scale_to_size:
+                    logger.debug("Could not continue to next page, truncating line.")
+                    line = line[: self.scale_to_size]
+
+        # Check if we should start a new page or continue the line on the current one
+        if self.max_lines is not None and self._line_count >= self.max_lines:
```

Comment:
This will also handle if `self.max_lines` is `0`, treating 0  as None
```suggestion
        if self.max_lines and self._line_count >= self.max_lines:
```